gateway/src/resty/http/request_reader.lua

```diff
@@ -0,0 +1,47 @@
+local httpc = require "resty.resolver.http"
+
+local _M = {
+}
+
+local cr_lf = "\r\n"
+
+-- chunked_reader return a body reader that translates the data read from
+-- lua-resty-http client_body_reader to HTTP "chunked" format before returning it
+--
+-- The chunked reader return nil when the final 0-length chunk is read
+local function chunked_reader(sock, chunksize)
+    chunksize = chunksize or 65536
+    local eof = false
+    local reader = httpc:get_client_body_reader(chunksize, sock)
```

Comment:
which method is `httpc:get_client_body_reader`?  I cannot find it

Reply:
if I read the code correct, we overwrite the lua-resty-http connect method in resty/resolver/http.lua and set ```
local _M = setmetatable({}, { __index = resty_http })
```

so `httpc:get_client_body_reader` will return the method from lua-resty-http lib.

Reply:
Why not require `local resty_http = require 'resty.http'` instead to make more explicit? the `get_client_body_reader` is a public exposed global function, not a method of any module table. Whatever you prefer.